nncf/quantization/algorithms/weight_compression/quantize.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Tuple
+
+import numpy as np
+
+from nncf.parameters import CompressWeightsMode
+from nncf.quantization.algorithms.weight_compression.compression_info import WeightCompressionConfig
+from nncf.quantization.fake_quantize import calculate_scale_zero_point
+
+
+def _do_integer_quantization(
```

Comment:
As functions in this file are used across nncf right now, I think it is possible to make it public
```suggestion
def do_integer_quantization(
```